pkg/kv/kvserver/raftentry/cache.go

```diff
@@ -258,6 +258,33 @@ func (c *Cache) Scan(
 	return ents, bytes, nextIdx, exceededMaxBytes
 }
 
+// ScanPartial is like Scan, but it may skip a prefix of entries that are not
+// cached. If the cached entries are [first, last+1), it scans a prefix of
+// entries in the [max(first, lo), min(last+1, hi)) interval, if any.
+func (c *Cache) ScanPartial(
```

Comment:
The `Scan` method is now unused (except in tests). But removing it and having only `ScanPartial` would be strange. Maybe find a way to implement one via the other.